addons/mrp/models/stock_move.py

```diff
@@ -160,6 +160,35 @@ def default_get(self, fields_list):
                 defaults['additional'] = True
         return defaults
 
+    def write(self, values):
+        if 'product_uom_qty' in values:
+            old_quantities = {move: move.product_uom_qty for move in self}
+        res = super().write(values)
+        if 'product_uom_qty' in values:
+            procurement_requests = []
+            for move in self:
+                mo = move.raw_material_production_id
+                if mo and mo.state == 'confirmed':
+                    qty_diff = float_compare(move.product_uom_qty, old_quantities[move], precision_rounding=move.product_uom.rounding)
+                    if qty_diff < 0:
+                        move.move_orig_ids._decrease_initial_demand(old_quantities[move] - move.product_uom_qty)
+            self.env['procurement.group'].run(procurement_requests)
+        return res
+
+    @api.model
+    def create(self, values):
+        if values.get('raw_material_production_id'):
+            mo = self.env['mrp.production'].browse(values['raw_material_production_id'])
+            values.update({
+                'group_id': mo.procurement_group_id.id,
+                'reference': mo.name
+            })
+        res = super().create(values)
+        if res.raw_material_production_id and res.raw_material_production_id.state == 'confirmed' and not res.bom_line_id:
+            res._adjust_procure_method()
+            res._action_confirm()
+        return res
```

Comment:
not sure i understand why you need this create override?

Reply:
I saw this was missing while testing flexible with propagate task.

Reply:
another commit then :)